### PR TITLE
Skip environment check if config is from core

### PR DESCRIPTION
**Pull request by terrytangyuan**
Since core `RunConfig` doesn't have `environment`.

### CODE REVIEW

**Comment by terrytangyuan** [comment]
Any update on the review of this yet? Would be nice to get this fix in before the release.

**Comment by terrytangyuan** [comment]
Oh just saw the conflicts. Seems like more elegant fix from internal has been pushed. Closing now.

**Comment by terrytangyuan** [comment]
I re-opened this since it doesn't to be fixed in 1.3.0. Resolved conflicts. Could anyone take a look?

**Comment by martinwicke** [comment]
xiejw@ can you take a look?

**Comment by martinwicke** [comment]
@terrytangyuan is it not fixed because the fix didn't make it into 1.3?

**Comment by terrytangyuan** [comment]
@martinwicke There was some changes in the same location but I realized later it was a fix for some other issue instead of this.

**Review by xiejw — COMMENTED** [review]
Sorry, I left the comments week ago and forgot to click the submit review button.

File: tensorflow/contrib/learn/python/learn/experiment.py [hunk 1]
```diff
@@ -318,15 +318,16 @@ def train(self, delay_secs=None):
...
+    if isinstance(config, run_config.RunConfig):
+      if (config.cluster_spec and config.master and
+          config.environment == run_config.Environment.LOCAL):
+        logging.warn("ClusterSpec and master are provided, but environment is "
+                     "set to 'local'. Set environment to 'cloud' if you intend "
+                     "to use the distributed runtime.")
+      if (config.environment != run_config.Environment.LOCAL and
+          config.environment != run_config.Environment.GOOGLE and
+          config.cluster_spec and config.master):
+        self._start_server()
```
[the remark this answers is not in the document]
Comment (terrytangyuan) [reply tensorflow/contrib/learn/python/learn/experiment.py]: Thanks! I just added change that reflect your second suggestion.

**Review by xiejw — APPROVED** [review]
Thanks.

**Comment by drpngx** [comment]
Jenkins, test this please.

**Comment by terrytangyuan** [comment]
Test failure seems unrelated?

**Comment by drpngx** [comment]
Yes, it's a transient failure that we get now and then. Since we want to test against windows still, I'll try again.

Jenkins, test this please.

**Pull request merged**